src/AzModuleInstaller.ts

```diff
@@ -0,0 +1,159 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+import * as os from 'os';
+import { ArchiveTools } from './Utilities/ArchiveTools';
+import FileUtils from './Utilities/FileUtils';
+import Utils from './Utilities/Utils';
+
+interface InstallResult {
+    moduleSource: string;
+    isInstalled: boolean;
+}
+
+export class AzModuleInstaller {
+    private version: string;
+    private githubAuth: string;
+    private moduleContainerPath: string;
+    private modulePath: string;
+    private moduleZipPath: string;
+    private installResult: InstallResult;
+    private isWin = false;
+
+    public constructor(version: string, githubAuth?: string) {
+        this.version = version;
+        this.githubAuth = githubAuth;
+        this.installResult = {
+            moduleSource: "Others",
+            isInstalled: false
+        };
+        const platform = (process.env.RUNNER_OS || os.type())?.toLowerCase();
+        switch(platform) {
+            case "windows":
+            case "windows_nt":
+                this.isWin = true;
+                this.moduleContainerPath = "C:\\Modules";
+                this.modulePath = `${this.moduleContainerPath}\\az_${this.version}`
+                break;
+            case "linux":
+                this.moduleContainerPath = "/usr/share";
+                this.modulePath = `${this.moduleContainerPath}/az_${this.version}`
+                break;
```

Comment:
no macos support ?

Reply:
This is the existing implementation, not sure though why it is, please have a look at the existing code in `src/Utilities:setPSModulePath`